src/app/auth/login/page.tsx

```diff
@@ -2,19 +2,28 @@
 
 import { Box } from '@primer/react'
 import { Login } from 'app/components/login/Login'
+import { useOrgsData } from 'hooks/useOrganizations'
 import { useSession } from 'next-auth/react'
 import { useRouter } from 'next/navigation'
 import { useEffect } from 'react'
 
 const LoginPage = () => {
   const session = useSession()
+  const orgsData = useOrgsData()
+
   const router = useRouter()
 
   useEffect(() => {
     if (session.data?.user) {
+      // redirect to org page if user has only one org
+      if (orgsData && orgsData.length === 1) {
```

Comment:
This is a race condition between our auth checker and the github API. If auth resolves first (which it usually will since its just a simple jwt decode), we'll always redirect to the org selection page, regardless of if they have orgs or not. Take a peep at these logs when I visit the page after signing in to get a better understanding of what's going on

```
{sessionState: 'authenticated', user: true, orgsDataLength: undefined}
{sessionState: 'authenticated', user: true, orgsDataLength: undefined}
{sessionState: 'authenticated', user: true, orgsDataLength: 1}
{sessionState: 'authenticated', user: true, orgsDataLength: 1}
{sessionState: 'authenticated', user: true, orgsDataLength: 1}
{sessionState: 'authenticated', user: true, orgsDataLength: 1}
```